jax/_src/nn/initializers.py

```diff
@@ -29,9 +30,26 @@
 from jax._src.util import prod
 from jax._src import dtypes
 
-DType = Any
-
-def zeros(key, shape, dtype: DType = jnp.float_):
+KeyArray = random.KeyArray
+Array = Any
+Shape = Tuple[int, ...]
```

Comment:
I think restricting shapes to ints is too restrictive, because shapes can be dynamic. e.g. `zeros` below calls `jnp.zeros`, which in turn calls `core.canonicalize_shape`, which uses this annotation: https://github.com/google/jax/blob/bad9cd57f0ecbf125732252e172d80bdef30b5bb/jax/core.py#L1571-L1575

Reply:
Thanks for catching this. I replaced all occurrences of `Shape` with `core.Shape`.